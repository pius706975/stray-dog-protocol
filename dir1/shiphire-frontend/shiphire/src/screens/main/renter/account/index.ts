export {default as RenterAccount} from './Account';
