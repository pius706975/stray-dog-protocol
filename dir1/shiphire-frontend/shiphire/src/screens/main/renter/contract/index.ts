export { default as ContractPreview } from "./ContractPreview"