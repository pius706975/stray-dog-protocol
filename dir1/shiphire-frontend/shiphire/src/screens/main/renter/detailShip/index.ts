export { default as DetailShip } from "./DetailShip"