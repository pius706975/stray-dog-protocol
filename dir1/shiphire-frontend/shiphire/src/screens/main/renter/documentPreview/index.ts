export {default as RenterDocumentPreview} from './RenterDocumentPreview';
