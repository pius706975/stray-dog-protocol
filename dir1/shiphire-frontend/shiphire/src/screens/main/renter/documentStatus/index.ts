export { default as RequestDetail } from './RequestDetail';
export { default as ProposalDetail } from './ProposalDetail';
export { default as ContractDetail } from './ContractDetail';
