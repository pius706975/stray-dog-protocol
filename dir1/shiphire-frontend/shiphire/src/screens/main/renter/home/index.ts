export { default as RenterHome } from "./Home"