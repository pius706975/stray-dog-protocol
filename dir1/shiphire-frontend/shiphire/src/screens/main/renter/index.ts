export * from './home';
export * from './notification';
export * from './account';
export * from './editProfile';
export * from './aboutUs';
export * from './savedShips';
export * from './verifyEmail';
export * from './renterCompany';
export * from './documentStatus';
export * from './detailShip';
export * from './search';
export * from './shipByCategory';
export * from './trasanctionStatus';
export * from './transactionDetail';
export * from './shipTrackingDetail';
export * from './shipTrackingForm';
export * from './requestForQuote';
export * from './rfqDocPreview';
export * from './proposal';
export * from './payment';
export * from './paymentHistory';
export * from './paymentReceipt';
export * from './contract';
export * from './trackTransactionsStat';
export * from './documentPreview';
export * from './verifyPhone';
export * from './remindedShips';
export * from './shipAvailability';
export * from './negotiate';
export * from './deleteAccount'
export * from './accountVerification'