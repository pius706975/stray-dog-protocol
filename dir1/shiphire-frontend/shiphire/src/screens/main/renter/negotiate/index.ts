export { default as Negotiate } from './Negotiate';
