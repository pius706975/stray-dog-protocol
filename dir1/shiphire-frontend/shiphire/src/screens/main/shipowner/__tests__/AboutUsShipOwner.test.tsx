import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { NavigationContainer } from '@react-navigation/native';
import {
    fireEvent,
    render,
    screen,
    waitFor,
} from '@testing-library/react-native';
import MockAdapter from 'axios-mock-adapter';
import { QueryClient, QueryClientProvider } from 'react-query';
import { Provider } from 'react-redux';
import httpRequest from '../../../../services/api';
import store from '../../../../store';
import {
    MainOwnerStackParamList,
    MainScreenOwnerParamList,
} from '../../../../types';
import { AboutUs } from '../../renter';

const queryClient = new QueryClient();
const MainScreenStack = createNativeStackNavigator<MainScreenOwnerParamList>();
const MainOwnerStack = createNativeStackNavigator<MainOwnerStackParamList>();

const mockAdapter = new MockAdapter(httpRequest);

const aboutUsMockComponent = (
    <QueryClientProvider client={queryClient}>
        <Provider store={store}>
            <NavigationContainer>
                <MainScreenStack.Navigator>
                    <MainOwnerStack.Screen name="AboutUs" component={AboutUs} />
                </MainScreenStack.Navigator>
            </NavigationContainer>
        </Provider>
    </QueryClientProvider>
);

describe('Testing About Us Screen', () => {
    describe('Snapshot Testing', () => {
        it('should render About Us Screen correctly', async () => {
            const aboutUsScreen = render(aboutUsMockComponent);
            expect(aboutUsScreen).toMatchSnapshot();
        });
    });
    describe('Rendering Testing', () => {
        it('should render text welcome correctly', async () => {
            const { getByText } = render(aboutUsMockComponent);
            const textWelcome = getByText('AboutUs.textWelcome');
            expect(textWelcome).toBeTruthy();
        });
        it('should render text title our mission correctly', async () => {
            const { getByText } = render(aboutUsMockComponent);
            const textTitleOurMission = getByText(
                'AboutUs.textTitleOurMission',
            );
            expect(textTitleOurMission).toBeTruthy();
        });
        it('should render text content our mission correctly', async () => {
            const { getByText } = render(aboutUsMockComponent);
            const textContentOurMission = getByText(
                'AboutUs.textContentOurMission',
            );
            expect(textContentOurMission).toBeTruthy();
        });
    });
});
