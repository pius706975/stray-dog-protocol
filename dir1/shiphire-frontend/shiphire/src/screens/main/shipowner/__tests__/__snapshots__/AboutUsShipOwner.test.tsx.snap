// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing About Us Screen Snapshot Testing should render About Us Screen correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    {
      "flex": 1,
    }
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onGestureCancel={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      sheetAllowedDetents="large"
      sheetCornerRadius={-1}
      sheetExpandsWhenScrolledToEdge={true}
      sheetGrabberVisible={false}
      sheetLargestUndimmedDetent="all"
      stackPresentation="push"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          collapsable={false}
          style={
            [
              {
                "flex": 1,
              },
              {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              [
                undefined,
                undefined,
                {
                  "flex": 1,
                },
                undefined,
                {},
                {},
                {},
                {
                  "backgroundColor": "#f0f6fb",
                  "justifyContent": "space-between",
                  "position": "relative",
                },
              ]
            }
            testID="AboutUs"
          >
            <RCTScrollView
              contentContainerStyle={
                {
                  "alignItems": undefined,
                  "flex": undefined,
                  "flexGrow": 1,
                  "gap": undefined,
                  "justifyContent": undefined,
                  "marginBottom": undefined,
                  "padding": 20,
                  "paddingVertical": undefined,
                }
              }
            >
              <View>
                <View
                  style={
                    [
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      {
                        "alignItems": "flex-start",
                      },
                    ]
                  }
                >
                  <Text
                    style={
                      [
                        {
                          "backgroundColor": "transparent",
                          "color": "#20303C",
                          "writingDirection": "ltr",
                        },
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        {
                          "fontFamily": "Kanit-Regular",
                          "fontSize": 16,
                        },
                      ]
                    }
                  >
                    AboutUs.textWelcome
                  </Text>
                </View>
                <View
                  style={
                    [
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      {
                        "alignItems": "center",
                      },
                    ]
                  }
                >
                  <Text
                    style={
                      [
                        {
                          "backgroundColor": "transparent",
                          "color": "#20303C",
                          "writingDirection": "ltr",
                        },
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        {
                          "color": "#216dab",
                          "fontFamily": "Kanit-Bold",
                          "fontSize": 22,
                        },
                      ]
                    }
                  >
                    AboutUs.textTitleOurMission
                  </Text>
                  <View
                    style={
                      [
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        {
                          "alignItems": "flex-start",
                        },
                      ]
                    }
                  >
                    <Text
                      style={
                        [
                          {
                            "backgroundColor": "transparent",
                            "color": "#20303C",
                            "writingDirection": "ltr",
                          },
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          {
                            "fontFamily": "Kanit-Regular",
                            "fontSize": 16,
                          },
                        ]
                      }
                    >
                      AboutUs.textContentOurMission
                    </Text>
                  </View>
                </View>
                <View
                  style={
                    [
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      {
                        "alignItems": "center",
                      },
                    ]
                  }
                >
                  <Text
                    style={
                      [
                        {
                          "backgroundColor": "transparent",
                          "color": "#20303C",
                          "writingDirection": "ltr",
                        },
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        {
                          "color": "#216dab",
                          "fontFamily": "Kanit-Bold",
                          "fontSize": 22,
                        },
                      ]
                    }
                  >
                    AboutUs.textTitleOurValues
                  </Text>
                  <View
                    style={
                      [
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        {
                          "alignItems": "flex-start",
                        },
                      ]
                    }
                  >
                    <Text
                      style={
                        [
                          {
                            "backgroundColor": "transparent",
                            "color": "#20303C",
                            "writingDirection": "ltr",
                          },
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          {
                            "fontFamily": "Kanit-Regular",
                            "fontSize": 16,
                          },
                        ]
                      }
                    >
                      AboutUs.textContentOurValues1
                    </Text>
                    <Text
                      style={
                        [
                          {
                            "backgroundColor": "transparent",
                            "color": "#20303C",
                            "writingDirection": "ltr",
                          },
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          {
                            "fontFamily": "Kanit-Regular",
                            "fontSize": 16,
                          },
                        ]
                      }
                    >
                      AboutUs.textContentOurValues2
                    </Text>
                    <Text
                      style={
                        [
                          {
                            "backgroundColor": "transparent",
                            "color": "#20303C",
                            "writingDirection": "ltr",
                          },
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          {
                            "fontFamily": "Kanit-Regular",
                            "fontSize": 16,
                          },
                        ]
                      }
                    >
                      AboutUs.textContentOurValues3
                    </Text>
                    <Text
                      style={
                        [
                          {
                            "backgroundColor": "transparent",
                            "color": "#20303C",
                            "writingDirection": "ltr",
                          },
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          {
                            "fontFamily": "Kanit-Regular",
                            "fontSize": 16,
                          },
                        ]
                      }
                    >
                      AboutUs.textContentOurValues4
                    </Text>
                  </View>
                </View>
                <View
                  style={
                    [
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      {
                        "alignItems": "center",
                      },
                    ]
                  }
                >
                  <Text
                    style={
                      [
                        {
                          "backgroundColor": "transparent",
                          "color": "#20303C",
                          "writingDirection": "ltr",
                        },
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        {
                          "color": "#216dab",
                          "fontFamily": "Kanit-Bold",
                          "fontSize": 22,
                        },
                      ]
                    }
                  >
                    AboutUs.textTitleGetInTouch
                  </Text>
                  <View
                    style={
                      [
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        {
                          "alignItems": "flex-start",
                        },
                      ]
                    }
                  >
                    <Text
                      style={
                        [
                          {
                            "backgroundColor": "transparent",
                            "color": "#20303C",
                            "writingDirection": "ltr",
                          },
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          {
                            "fontFamily": "Kanit-Regular",
                            "fontSize": 16,
                          },
                        ]
                      }
                    >
                      AboutUs.textContentGetInTouch
                      <Text
                        style={
                          [
                            {
                              "backgroundColor": "transparent",
                              "color": "#20303C",
                              "writingDirection": "ltr",
                            },
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            {
                              "color": "#216dab",
                              "fontFamily": "Kanit-Regular",
                              "fontSize": 16,
                            },
                          ]
                        }
                      >
                         
                        support@shiphire.com.
                      </Text>
                    </Text>
                  </View>
                </View>
                <View
                  style={
                    [
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      {
                        "alignSelf": "center",
                      },
                    ]
                  }
                >
                  <Text
                    style={
                      [
                        {
                          "backgroundColor": "transparent",
                          "color": "#20303C",
                          "writingDirection": "ltr",
                        },
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        {
                          "color": "#216dab",
                          "fontFamily": "Kanit-Regular",
                          "fontSize": 18,
                        },
                      ]
                    }
                  >
                    ©
                    ShipHire 2023
                  </Text>
                </View>
              </View>
            </RCTScrollView>
          </View>
        </View>
      </View>
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backTitleVisible={true}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="AboutUs"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
